snap/frontend/components/TransferModalBox/index.jsx

```diff
@@ -0,0 +1,274 @@
+import { useWalletContext } from '../../context';
+import Button from '../Button';
+import Input from '../Input';
+import ModalBox from '../ModalBox';
+import { useEffect, useState } from 'react';
+import { SymbolFacade } from 'symbol-sdk/symbol';
+
+const TransferModalBox = ({ isOpen, onRequestClose }) => {
+	const { walletState } = useWalletContext();
+	const { selectedAccount, network, mosaicInfo } = walletState;
+	const mosaicsBalance = selectedAccount.mosaics || [];
+	const facade = new SymbolFacade(network.networkName);
+
+	const [to, setTo] = useState('');
+	const [message, setMessage] = useState('');
+	const [isEncrypt, setIsEncrypt] = useState(false);
+	const [mosaicSelectorManager, setMosaicSelectorManager] = useState([]);
+	const [selectedMosaics, setSelectedMosaics] = useState([]);
+	const [errors, setErrors] = useState({
+		address: null,
+		message: null,
+		amount: []
+	});
+
+	const validate = (field, value) => {
+		switch (field) {
+		case 'address':
+			return facade.network.isValidAddressString(value) ? null : 'Invalid address';
+		case 'message':
+			return 1024 >= value.length ? null : 'Message length should not exceed 1024 characters';
+		case 'amount':
+			return value.map((amount, index) => {
+				const maxBalance = mosaicsBalance.find(mosaic => mosaic.id === selectedMosaics[index].id).amount;
+				const absoluteAmount = amount * (10 ** mosaicInfo[selectedMosaics[index].id].divisibility);
+
+				return absoluteAmount > maxBalance ? 'Not enough balance' : null;
+			});
+		default:
+			return null;
+		}
+	};
+
+	const validateForm = () => {
+		const validate = (field, value) => {
```

Comment:
This validate function looks the same as above function?

Reply:
good catch! 👍🏼